active/0000-app-import-output-file-for-addons.md

```diff
@@ -0,0 +1,116 @@
+- Start Date: 2016-04-03
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+Allows `app.import` to specify outputFile for addons.
+Not calling `app.import` for an addon would be considered to have
+an outputFile of `assets/vendor.js` to maintain the existing behavior.
+
+# Motivation
+
+[RFC#28](https://github.com/ember-cli/rfcs/pull/28) added support to specifiy an `outputFile` this works great for static `vendor` files, however, the addon's processed JS always end up in `vendor.js`. We have the same motiviation to do this for addons as we do for other vendor assets. This is specially important for large applications where only parts of the app depend on certain addons. Splitting the output vendor files allows us to cache assets more efficiently or even lazy-load them. Note: lazy-load is out of the scope of this RFC
+
+
+# Detailed design
+
+Today the addonTree processes JS, which get contenated together to `addons.js` and later to `vendor.css` and `vendor.js` respectively. With this change, we will allow the addon consumers further control about where the processed JS output for each addon ends up. Today we don't process addon's CSS directly this way, but we already have a solution via `outputPaths`, for this proposal CSS is ignored. See alternatives for a consistent option with CSS and JS.
```

Comment:
ya, it seems like the @chadhietala work merely add extra sanity to this. So it could be layered on after (at-least it seems like it)